File: client/src/test/java/com/jraska/github/client/ui/UsersActivityTest.java
package com.jraska.github.client.ui;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(RobolectricTestRunner.class)
public class UsersActivityTest {
  @Test
  public void whenCreated_thenUsersRecyclerSetUp() throws Exception {
    UsersActivity usersActivity = Robolectric.setupActivity(UsersActivity.class);

    assertThat(usersActivity.usersRecyclerView).isNotNull();
  }
}

File: feature/users/src/androidTest/java/com/jraska/github/client/users/test/UsersActivityTest.kt
package com.jraska.github.client.users.test

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.rule.ActivityTestRule
import com.jraska.github.client.android.test.http.assetJson
import com.jraska.github.client.android.test.usingLinkRecording
import com.jraska.github.client.http.MockWebServerInterceptorRule
import com.jraska.github.client.users.ui.UsersActivity
import okhttp3.mockwebserver.MockResponse
import okhttp3.mockwebserver.MockWebServer
import org.assertj.core.api.Assertions.assertThat
import org.junit.Rule
import org.junit.Test

class UsersActivityTest {

  @get:Rule
  val mockWebServer = MockWebServer()

  @get:Rule
  val mockWebServerInterceptorRule = MockWebServerInterceptorRule(mockWebServer)

  @get:Rule
  val rule = ActivityTestRule(UsersActivity::class.java)

  @Test
  fun testLaunches() {
    mockWebServer.enqueue(twoUsersResponse())

    usingLinkRecording {
      onView(withText("defunkt")).perform(click())

      val lastScreen = it.linksLaunched.last()
      assertThat(lastScreen.toString()).isEqualTo("https://github.com/defunkt")

      onView(withText("mojombo")).perform(click())
      val nextScreen = it.linksLaunched.last()
      assertThat(nextScreen.toString()).isEqualTo("https://github.com/mojombo")
    }
  }

  private fun twoUsersResponse(): MockResponse {
    return assetJson("users/two_users.json")
  }
}
